### DISCUSSION TITLE
How to change context value of Parent page Layout

### DISCUSSION THREAD

**Discussion by TristanKirwan** [vercel/next.js#51301] [App Router]
Hi everyone.

Currently I am running into the following issue regarding Server-side components & Context. We have a multilingual project, with several nested routes using the App Router, like so:

```typescript
[lang]
page.tsx
layout.tsx
  [landing_uid]
   page.tsx
    [detail_uid]
     page.tsx
``` 
In the `layout` of the `[lang]` route, we render the header, footer and pass along a context provider that provides a glossary object to the children, like so:

```typescript
return (
    <GlossaryProvider lang={lang}>
        <Header {...navigationData} />
        {children}
        <Footer {...navigationData} />
    </GlossaryProvider>
  );
``` 

Inside the `header` component, there is another component `LanguageToggle`. I would like to pass the correct url for the translated version of the page the user is currently on. For context: `/en/news` and `/nl/nieuws` could be same page in the CMS and just be translations of the same page. I get this translated url in the `page.tsx` of the route the user is on.

The problem is that in this route I do not have access to the `/[lang]/layout.tsx` file, where I would like to pass this value via another Context provider. I also do not know the route on which I fetch the required data (for example, it could be on `[lang]/[landing_uid]` but it could also be on `[lang]/[landing_uid]/[detail_uid]`.

Previous I have used Remix, where I could use `useLoaderData` and get the relevant data straight in the component where i need it (`<LanguageToggle />`), but as far as I can see this is not available. I have also used a getter/setter method in combination with a class, but this doesn't work, as the `/[lang]/layout.tsx` would be hit before the actual page where I am doing the fetch call is being hit, meaning I don't have the data when I am setting it to the context provider in `/[lang]/layout.tsx`. I have also tried to update the context directly on my relevant page (`/[lang]/[landing_uid]`), but this is not possible on SSC.

Has anybody ran into a similar issue that could point me in the right direction?

**Comment by lamhungypl** [comment]
Hello, did you find a solution instead of prop drilling?
